OpenDreamServer/Dream/DreamMap.cs

```diff
@@ -99,7 +99,10 @@ public class MapLevel {
 
         public void RemoveLevel() {
             MapLevel toRemove = Levels[^1];
-            toRemove.Cells = null;
+            foreach (MapCell cell in toRemove.Cells) {
+                cell.Turf.Delete();
+                cell.Area.Delete();
```

Comment:
This will delete an area, even if it still contains other turfs or is referenced somewhere in the DM code.

Reply:
Here's what I have - can't find a way to compensate for higher z-levels though: 
```cs
public void RemoveLevel() {
    int idx = Levels.Count - 1;
    MapLevel toRemove = Levels[idx];
    foreach (MapCell cell in toRemove.Cells) {
        cell.Turf.Delete();
        // Area.z represents the lowest Z level a comprised turf is on, therefore we can delete if it's >= our idx
        // DOES NOT CONSIDER TURFS ON A HIGHER Z-LEVEL!
        if (cell.Area.GetVariable("z").GetValueAsInteger() >= idx) 
            cell.Area.Delete();
    }
    Levels.Remove(toRemove);
}
```

Reply:
Using the following code to test, it doesn't appear that the area _should_ be deleted after all of its turfs are gone. Its `x` `y` and `z` become null, but that's not something we can replicate without keeping track of an area's turfs, which we don't currently do.

```dm
world.maxz++

var/area/newarea = new /area/test()
newarea.contents += locate(1, 1, world.maxz)

world << "[newarea], [newarea.x], [newarea.y], [newarea.z]"
world.maxz--
world << "[newarea], [newarea.x], [newarea.y], [newarea.z]"
```

Output:
```
The test, 1, 1, 2
The test, , , 
```